doc/fwutil/fwutil.md

```diff
@@ -490,6 +540,24 @@ Aborted!
 
 ###### Figure 5: FW install (modular) flow
 
+## 3.4 Update component FW
+
+### 3.4.1 Non modular chassis platform
+
+TBD
+
+### 3.4.2 Modular chassis platform
+
+TBD
+
+### 3.4.3 Auto FW update based on the configuration file
+
+1. Find the available firmware based on boot type and immediate action type from platform specific fw update configuration file. 
+   Exit if no configuration file exists.
+2. Update the firmware using the script if it's specified in the configuration. Otherwise, fwutil will use the platform api to update the firmware.
+   Exit if the update fails in any step.
```

Comment:
@sujinmkang I would like to propose a new CLI tree:
```
fwutil
|--- show
|    |--- version
|    |--- status
|    |--- updates -i|--image=<current|next> -c|--config=<path_to_platform_fw_updates>
|
|--- install
|    |--- chassis
|    |    |--- component <component_name>
|    |         |--- fw -y|--yes <fw_path>
|    |
|    |--- module <module_name>
|         |--- component <component_name>
|              |--- fw -y|--yes <fw_path>
|
|--- update
     |--- chassis
     |    |--- component <component_name>
     |         |--- fw -y|--yes -f|--force -i|--image=<current|next> -c|--config=<path_to_platform_fw_updates>
     |
     |--- module <module_name>
     |    |--- component <component_name>
     |         |--- fw -y|--yes -f|--force -i|--image=<current|next> -c|--config=<path_to_platform_fw_updates>
     |
     |--- components
          |--- fw -y|--yes -f|--force -i|--image=<current|next> -b|--boot=<cold|fast|warm> -c|--config=<path_to_platform_fw_updates>
```

Reply:
@sujinmkang Some thoughts:
1. In case platform API is not implemented (plugin mode) the fwutil CLI won't be fully available - only automatic installation mode. That's something which need to be highlighted in this design.
2. Platform API should be extended with the new boot argument.
3. How to handle `version` and `script` fields in case Platform API is not implemented